### PR TITLE
ci: Combine coverage from all shards

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Allow shards to generate coverage outside of their domain
- Merge coverage reports from all shards and check overall threshold, failing when necessary

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]
🤔 why not uploading the merged coverage?

**Comment by zharinov** [comment]
Yeah, it's very frustrating, I have no idea why it doesn't upload `.txt`

**Comment by zharinov** [comment]
image

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -312,45 +312,62 @@ jobs:
       - name: Move coverage files
         if: (success() || failure()) && github.event.pull_request.draft != true && matrix.coverage
         run: |
-          mkdir ./coverage-reports
+          mkdir -p ./coverage/reports
           for shard in ${{ matrix.shards }};
           do
-            mv ./coverage/shard/$shard/lcov.info ./coverage-reports/$shard.txt
```
Comment (viceice) [inline .github/workflows/build.yml:318]: you probably need to leave lcov inside the name

see here for default fileMatch of codecov uploader

https://github.com/codecov/uploader/blob/main/src/helpers/files.ts#L161

Comment (zharinov) [reply .github/workflows/build.yml:318]: 🤦‍♂️

**Review by viceice — COMMENTED** [review]
see my comment about the supported upload files. you said lcov is better, so we should stay

**Comment by zharinov** [comment]
I'm not 100% sure it's better, but what I've found is that `nyc` tool uses it in their [Codecov integration guide](https://github.com/istanbuljs/nyc/blob/master/docs/setup-codecov.md)

**Pull request merged**